### ISSUE TITLE
win32 helper dies with limited logging

### ISSUE THREAD

**Issue by computergeek125** [vuori/weasel-pageant#5]
Wish I had more info for you on this one.  This process appears broken on the latest release of windows (installed yesterday).  When started, everything is normal, but when I attempt to connect to an SSH session, this happens:
```
me@des-dinosaur:~/winhome/src/weasel-pageant$ ./weasel-pageant -sd
SSH_AUTH_SOCK='/tmp/ssh-9MZedc/agent.129'; export SSH_AUTH_SOCK;
SSH_PAGEANT_PID=129; export SSH_PAGEANT_PID;
echo ssh-pageant pid 129;
/mnt/c/Users/me/src/weasel-pageant-1.1/helper.exe: Invalid argument
weasel-pageant: win32 helper died immediately
```

This occurs both when manually starting in debug mode (eval strings copy/pasted over to the new window) or when in the same window started via `.profile`.

My current Windows 10 version is 1803 build 17692.1000 (Insider Preview).  The folder name is accurate; I am running version 1.1

**Comment by vuori** [comment]
Yeah, the release binary has the wrong version number. Can you run other win32 programs (e.g. `ipconfig.exe`)? What happens if you try to run `helper.exe` manually from shell?

**Comment by computergeek125** [comment]
ipconfig works, I learned that cmd.exe can actually run inside gnome-terminal, but helper complains mildly

```shell
 ~/winhome/src/JudaiBot ▷ master ▷ 133d22h18m ▷ 
 ❯❯❯ ~/winhome/src/weasel-pageant/helper.exe
This program is part of weasel-pageant and cannot be executed directly.%

Last command exited 1!
```

**Comment by vuori** [comment]
That sounds pretty weird. I'm not running Insider builds anywhere right now so I can't test this. Can you try specifying the path to the helper binary manually with the `-H` option, for example `weasel-pageant -d -H ~/winhome/src/weasel-pageant/helper.exe` and see if it has any effect?

**Comment by computergeek125** [comment]
Unfortunately no change.  I also tried knocking out the symlinks in the path (`~/winhome` -> `/mnt/c/users/me`; `weasel-pageant` -> `weasel-pageant-1.1`), and downloading a new copy and running it from my lxfs home directory versus the drvfs mountpoint

**Comment by computergeek125** [comment]
To note: outside of wsl, putty is working with my ssh keys loaded

**Comment by vuori** [comment]
Is this with Ubuntu 18.04? Have you set any mount options in `wsl.conf`? The error looks like `posix_spawn` creates the child process but fails to exec it.

**Comment by computergeek125** [comment]
I was originally using 16.04, however this is still present in 18.04.  I did not have a wsl.conf, but I created one and the system appears to ignore the file completely.  Since that's probably a different (but possibly related) issue, I'm going to close this and call it a system quirk for now rather than a bug.  If it persists after fixing the wsl.conf, I'll reopen.

**Comment by vuori** [comment]
Please report if you find a solution. I probably have an underused Windows 10 dev license somewhere that I can throw into a VM for insider builds, but it's going to be a few weeks before I have time to wrangle with it.

**Comment by computergeek125** [comment]
All right.  I finally got my wsl.conf to behave (RIP copy/paste), so back to reproducible issue :/
```shell
ssh-pageant pid 57
 ~ ▷  ▷  ▷ 
 ❯❯❯ lsb_release -a                                                                                 ◁  
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.4 LTS
Release:	16.04
Codename:	xenial
 ~ ▷  ▷  ▷ 
 ❯❯❯ grep meta /etc/wsl.conf                                                                        ◁  
options = "metadata,umask=022,fmask=011"
 ~ ▷  ▷  ▷ 
 ❯❯❯ mount | grep meta                                                                              ◁  
C: on /mnt/c type drvfs (rw,noatime,uid=1000,gid=1000,umask=22,fmask=11,metadata)
F: on /mnt/f type drvfs (rw,noatime,uid=1000,gid=1000,umask=22,fmask=11,metadata)
 ~ ▷  ▷  ▷ 
 ❯❯❯ ssh srv-aten                                                                                   ◁  
/mnt/c/Users/me/src/weasel-pageant-1.1/helper.exe: Invalid argument
weasel-pageant: win32 helper died immediately
me@srv-aten's password: 

                                                                                                        
Last command exited 130!
 ~ ▷  ▷  ▷ 
 ❯❯❯ 
```

Similar results on the Ubuntu 18.04 package from canonical.

**Comment by vuori** [comment]
When run with the `-d` flag, the debug output is printed on stderr. What's failing is probably the `posix_spawn` call on line 323. According to the log the helper doesn't start for whatever reason, so no logging can be expected from there.

**Comment by vuori** [comment]
As far as I can tell this is fixed, but tracking in #6 for now.
